Common/AR/include/ArcGISArViewInterface.h

```diff
@@ -39,7 +40,10 @@ class ArcGISArViewInterface : public QQuickFramebufferObject
   Q_PROPERTY(bool tryUsingArKit READ tryUsingArKit WRITE setTryUsingArKit NOTIFY tryUsingArKitChanged)
 
   // sensor
-  Q_PROPERTY(LocationDataSource* locationDataSource READ locationDataSource WRITE setLocationDataSource NOTIFY locationDataSourceChanged)
+  Q_PROPERTY(LocationDataSource* locationDataSource READ locationDataSource
+             WRITE setLocationDataSource NOTIFY locationDataSourceChanged)
+  Q_PROPERTY(LocationTrackingMode locationTrackingMode READ locationTrackingMode
```

Comment:
Is this enum registered somewhere? That will allow it to be accessed in Qml code.

Reply:
I thought the same thing about to register the enum with `Q_ENUM`.
The problem is `Q_ENUM` must by called in a `QObject` or `QGadget` class. I'm trying to create the enum outside the class and to call `Q_ENUM` in a class `ArEnums`, but I'm not sure if it's possible.
```
// In LocationTrackingMode.h
enum class LocationTrackingMode
{
  Ignore = 0,
  Initial = 1,
  Continuous = 2
};

// In ArEnums.h
class ArEnums
{
  Q_GADGET

public:
  Q_ENUM(LocationTrackingMode)

  ArEnums();
  ~ArEnums();
};
```
In ArcGIS Apis, there are enums for C++ defined outside a class (which are not registered with `Q_ENUM`) and enums for QML defined inside the `QmlEnums` class (which are registered with `Q_ENUM`).

Do you think it's fine to move the definition of enums into the `ArEnums` class? (And then to change the use of this enums:)
```
LocationTrackingMode locationTrackingMode() const;
```
become
```
ArEnums::LocationTrackingMode locationTrackingMode() const;
```

Reply:
Or maybe to move the enums in `LocationDataSource` class?

cc @JamesMBallard

Reply:
@JamesMBallard I tried to move the enums in `LocationDataSource` and use `Q_ENUM` on them, but it doesn't work. Maybe I miss something.
```
qrc:/qml/main.qml:60: Error: Cannot assign int to an unregistered type
```

Reply:
@GuillaumeBelz you would also need to add `qRegisterMetaType` for each of the enums. And only one `Q_ENUM` declaration per enum.

Reply:
>Do you think it's fine to move the definition of enums into the ArEnums class?

Let's talk about this. We need to have the readability/usability/discoverability improved for Qml. We don't want to force users to pass ints.  The enums need to be first class citizens in this API.

Reply:
@JamesMBallard Before to commit, here the changes for APIs.
I moved the enum declaration in `ArEnum`
```
class ArEnums
{
  Q_GADGET

public:
  enum class LocationTrackingMode
  {
    Ignore = 0,
    Initial = 1,
    Continuous = 2
  };
  Q_ENUM(LocationTrackingMode)

  enum class SensorStatus
  {
    Stopped = 0,
    Starting = 1,
    Started = 2
  };
  Q_ENUM(SensorStatus)

  ArEnums();
  ~ArEnums();
};
```
In C++, the `ArEnums` must be added when the enum is used:
```
LocationTrackingMode::Initial
// become
ArEnums::LocationTrackingMode::Initial
```
In QML, the enums can be used using the uncreatable type `ArEnums`
```
ArEnums.Initial // means LocationTrackingMode::Initial
```
Do you think this is ok?
I don't like to have only `ArEnums.Initial` in QML, it's not very explicit. 
2 solutions, I guess:
# use more explicit name for the class

```
class LocationTrackingModeEnums
{
  Q_GADGET

public:
  enum class LocationTrackingMode
  {
    Ignore = 0,
    Initial = 1,
    Continuous = 2
  };
  Q_ENUM(LocationTrackingMode)

  // SensoreStatus must be in another class

  LocationTrackingModeEnums();
  ~LocationTrackingModeEnums();
};

// C++
LocationTrackingModeEnums::LocationTrackingMode::Ignore

// QML
LocationTrackingModeEnums.Ignore
```

# use more explicit name for the enum values
class ArEnums
{
  Q_GADGET

public:
  enum class LocationTrackingMode
  {
    LocationTrackingModeIgnore = 0,
    LocationTrackingModeInitial = 1,
    LocationTrackingModeContinuous = 2
  };
  Q_ENUM(LocationTrackingMode)

  enum class SensorStatus
  {
    SensorStatusStopped = 0,
    SensorStatusStarting = 1,
    SensorStatusStarted = 2
  };
  Q_ENUM(SensorStatus)

  LocationTrackingModeEnums();
  ~LocationTrackingModeEnums();
};

// C++
LocationTrackingMode::LocationTrackingModeIgnore

// QML
ArEnums.LocationTrackingModeIgnore
```

Reply:
@GuillaumeBelz let's go with the second option: `use more explicit name for the enum values`. That is a bit redundant for C++ code, but it's much better for Qml readability and overall code layout.